### PR TITLE
Add version management

### PR DESCRIPTION
**Pull request by bCoder778**

### CODE REVIEW

**Review by forchain — CHANGES_REQUESTED** [review]
cannot generate new build after removed
```build
 rm build/bin/qitmeer-wallet
make
```

make: `build' is up to date.

**Review by forchain — COMMENTED** [review]
One suggestion

By your current implementation , we cannot view qitmeer version and the binary location by make command as Qitmeer does. 
```
Done building.
  qitmeer version 0.9.2+dev-ac02016 (Go version go1.16.3))
Run "./build/bin/qitmeer" to launch.

```

So if I made a build before, there is no way for me to find where the build is.
```
make: `build' is up to date.

```

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,67 @@
...
+	build/release/linux/amd64/bin/$(EXECUTABLE)
+WIN_EXECUTABLES := \
+	build/release/windows/amd64/bin/$(EXECUTABLE).exe
+
+EXECUTABLES=$(UNIX_EXECUTABLES) $(WIN_EXECUTABLES)
+
+COMPRESSED_EXECUTABLES=$(UNIX_EXECUTABLES:%=%.tar.gz) $(WIN_EXECUTABLES:%.exe=%.zip) $(WIN_EXECUTABLES:%.exe=%.cn.zip)
+
+RELEASE_TARGETS=$(EXECUTABLES) $(COMPRESSED_EXECUTABLES)
+
```
Comment (lochjin) [inline Makefile:26]: `.PHONY: build`

Comment (lochjin) [inline Makefile:26]: .PHONY: build

**Review by lochjin — CHANGES_REQUESTED** [review]

**Review by dindinw — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+UNIX_EXECUTABLES := \
+	build/release/darwin/amd64/bin/$(EXECUTABLE) \
+	build/release/linux/amd64/bin/$(EXECUTABLE)
+WIN_EXECUTABLES := \
+	build/release/windows/amd64/bin/$(EXECUTABLE).exe
+
+EXECUTABLES=$(UNIX_EXECUTABLES) $(WIN_EXECUTABLES)
+
+COMPRESSED_EXECUTABLES=$(UNIX_EXECUTABLES:%=%.tar.gz) $(WIN_EXECUTABLES:%.exe=%.zip) $(WIN_EXECUTABLES:%.exe=%.cn.zip)
```
[the remark this answers is not in the document]
Comment (bCoder778) [reply Makefile]: ok

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,67 @@
...
+	@GOOS=$(OS) GOARCH=$(ARCH) go build $(GOFLAGS_RELEASE) -o $(@) "github.com/Qitmeer/qitmeer-wallet"
+
+
+%.zip: %.exe
+	@echo zip $(EXECUTABLE)-$(VERSION)-$(OS)-$(ARCH)
+	@zip $(EXECUTABLE)-$(VERSION)-$(OS)-$(ARCH).zip "$<"
+%.cn.zip: %.exe
+	@echo Build $(@).cn.zip
+	@echo zip $(EXECUTABLE)-$(VERSION)-$(OS)-$(ARCH)
+	@zip -j $(EXECUTABLE)-$(VERSION)-$(OS)-$(ARCH).cn.zip "$<" script/win/start.bat
```
[the remark this answers is not in the document]
Comment (bCoder778) [reply Makefile]: ok

File: Makefile (hunk 1 above)
Comment (bCoder778) [reply Makefile:26]: I'm sorry, I didn't understand what you meant, could you elaborate

**Review by forchain — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (dindinw) [reply Makefile:26]: A `.PHONY` target is one that is not really the name of a file,  The `.PHONY` make sure `make` will run the recipe regardless looking for a file target. see https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html for more details.

**Review by dindinw — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (bCoder778) [reply Makefile:26]: ok

**Review by lochjin — APPROVED** [review]

**Pull request merged**